dbms/src/Flash/Planner/plans/PhysicalJoin.cpp

```diff
@@ -0,0 +1,269 @@
+
+    auto [other_condition_expr, other_filter_column_name, other_eq_filter_from_in_column_name]
+        = tiflash_join.genJoinOtherConditionAction(context, left_input_header, right_input_header, probe_side_prepare_actions);
+
+    const Settings & settings = context.getSettingsRef();
+    size_t max_block_size_for_cross_join = settings.max_block_size;
+    fiu_do_on(FailPoints::minimum_block_size_for_cross_join, { max_block_size_for_cross_join = 1; });
+
+    JoinPtr join_ptr = std::make_shared<Join>(
+        probe_key_names,
+        build_key_names,
+        true,
+        SizeLimits(settings.max_rows_in_join, settings.max_bytes_in_join, settings.join_overflow_mode),
+        tiflash_join.kind,
+        tiflash_join.strictness,
+        log->identifier(),
+        tiflash_join.join_key_collators,
+        probe_filter_column_name,
+        build_filter_column_name,
+        other_filter_column_name,
+        other_eq_filter_from_in_column_name,
+        other_condition_expr,
+        max_block_size_for_cross_join,
+        match_helper_name);
+
+    recordJoinExecuteInfo(dag_context, executor_id, build_plan->execId(), join_ptr);
+
+    auto physical_join = std::make_shared<PhysicalJoin>(
+        executor_id,
+        join_output_schema,
+        log->identifier(),
+        probe_plan,
+        build_plan,
+        join_ptr,
+        columns_added_by_join,
+        probe_side_prepare_actions,
+        build_side_prepare_actions,
+        is_tiflash_right_join,
+        PhysicalPlanHelper::constructBlockFromSchema(join_output_schema));
+    return physical_join;
+}
+
+void PhysicalJoin::probeSideTransform(DAGPipeline & probe_pipeline, Context & context)
+{
+    const auto & settings = context.getSettingsRef();
+    auto & dag_context = *context.getDAGContext();
+
+    /// probe side streams
+    executeExpression(probe_pipeline, probe_side_prepare_actions, log, "append join key and join filters for probe side");
+    auto join_probe_actions = PhysicalPlanHelper::newActions(probe_pipeline.firstStream()->getHeader(), context);
+    join_probe_actions->add(ExpressionAction::ordinaryJoin(join_ptr, columns_added_by_join));
+    /// add join input stream
+    if (has_non_joined)
+    {
+        auto & join_execute_info = dag_context.getJoinExecuteInfoMap()[execId()];
+        size_t not_joined_concurrency = join_ptr->getNotJoinedStreamConcurrency();
+        const auto & input_header = probe_pipeline.firstStream()->getHeader();
+        for (size_t i = 0; i < not_joined_concurrency; ++i)
+        {
+            auto non_joined_stream = join_ptr->createStreamWithNonJoinedRows(input_header, i, not_joined_concurrency, settings.max_block_size);
+            non_joined_stream->setExtraInfo("add stream with non_joined_data if full_or_right_join");
+            probe_pipeline.streams_with_non_joined_data.push_back(non_joined_stream);
+            join_execute_info.non_joined_streams.push_back(non_joined_stream);
+        }
```

Comment:
after merge master to planner_refactory
I will port the change from https://github.com/pingcap/tiflash/pull/5411

Reply:
done